plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/OpenShiftResourceUIModel.java

```diff
@@ -19,30 +21,41 @@
  * Display model for OpenShift resources to allow
  * overriding equals and hash
  * @author jeff.cantrill
+ * @author Jeff Maury
  *
  */
 public class OpenShiftResourceUIModel extends ObservableUIPojo implements IResourceUIModel{
 
-	private final IResource resource;
-	private final String tostring;
+	private IResource resource;
+	private String tostring;
 	private final Object parent;
+	private AtomicBoolean deleting = new AtomicBoolean(false);
+
 
-	public OpenShiftResourceUIModel(IResource resource, IResourcesUIModel parent){
+	public OpenShiftResourceUIModel(IResource resource, Object parent){
```

Comment:
OpenShiftResourceUIModel is the base class for all wrappers so why do you need another class ?